packages/react-dom/src/__tests__/ReactDOMComponentTree-test.js

```diff
@@ -68,52 +78,115 @@ describe('ReactDOMComponentTree', () => {
     expect(renderAndGetRef('input')).toBe('INPUT');
   });
 
-  it('finds instances for nodes', () => {
-    class Component extends React.Component {
+  it('finds closest instance for node when an event happens', done => {
+    const elemID = 'aID';
+    const innerHTML = {__html: `<div id="${elemID}"></div>`};
+
+    class ClosestInstance extends React.Component {
+      id = 'closestInstance';
+      _onClick = e => {
+        const node = e.currentTarget;
+        expect(node.id).toBe(this.id);
+        done();
+      };
       render() {
         return (
-          <div>
-            <h1>hello</h1>
-            <p>
-              <input />
-            </p>
-            goodbye.
-            <main dangerouslySetInnerHTML={{__html: '<b><img></b>'}} />
-          </div>
+          <div
+            id="closestInstance"
+            onClick={this._onClick}
+            dangerouslySetInnerHTML={innerHTML}
+          />
         );
       }
     }
 
-    function renderAndQuery(sel) {
-      var root = renderMarkupIntoDocument(<section><Component /></section>);
-      return sel ? root.querySelector(sel) : root;
-    }
+    const component = <ClosestInstance />;
+    const container = document.createElement('div');
+    ReactDOM.render(<section>{component}</section>, container);
+    document.body.appendChild(container);
+    simulateClick(document.getElementById(elemID));
+  });
 
-    function renderAndGetInstance(sel) {
-      return ReactDOMComponentTree.getInstanceFromNode(renderAndQuery(sel));
+  it('finds a controlled instance from node and gets its current fiber props', done => {
+    const inputID = 'inputID';
+    const startValue = undefined;
+    const finishValue = 'finish';
+
+    class Controlled extends React.Component {
+      state = {value: startValue};
+      a = null;
+      _onChange = e => {
+        const node = e.currentTarget;
+        expect(node.value).toEqual(finishValue);
```

Comment:
The problem with tests like this is that if `_onChange` never fires, the test passes (since none of the expectations were called). We need to move those out.

Reply:
Thanks for the comment @gaearon . I'll update to move assertions out of the event handler - this is much simpler. 

Though FWIW from my understanding of [async tests in Jest](https://facebook.github.io/jest/docs/en/asynchronous.html#callbacks) if `done()` is never called, due to `_onChange` not firing, then it would fail - albeit the failure message would not be the most helpful.